### PR TITLE
Traefik reverse proxy

### PR DESCRIPTION
**Pull request by svengo**
Add configuration for traefik, fixes mailcow/mailcow-dockerized#667. 
I think the only problem is that the user has to modify `docker-compose.yml`, it's not enough to change  `HTTP_BIND` and `HTTPS_BIND` to `127.0.0.1`.

### CODE REVIEW

File: docs/firststeps-rp.md [hunk 1]
```diff
@@ -79,3 +79,32 @@ backend mailcow
...
+      expose:
+         - "80"
+      labels:
+        - "traefik.backend=mailcow"
+        - "traefik.docker.network=traefik_web"
+        - "traefik.enable=true"
+        - "traefik.port=80"
+        - "traefik.frontend.rule=Host:${MAILCOW_HOSTNAME}"
+      networks:
+        web:
```
Comment (mkuron) [inline docs/firststeps-rp.md]: Is this even correct syntax? The line ends in a colon but there is no indented section on the next line.

Comment (svengo) [reply docs/firststeps-rp.md]: Mhh, the above syntax is accepted by `docker-compose config` but a dash would look better:

          networks:
            - web

**Comment by mkuron** [comment]
@svengo:
- Does Traefik set the `X-Forwarded-Proto` header automatically and preserve the `Host` header? If not, the CalDAV/CardDAV URLs you can copy from the SOGo interface may be wrong.
- If you are using a separate Docker network, the nginx server won't accept the `X-Forwarded-For` headers coming from Traefik. So `docker-compose logs nginx-mailcow` will show the same IP for every request and there might be some other problems.
- Please change the docker-compose syntax as you wrote.

**Comment by Braintelligence** [comment]
@mkuron Thank you for your observations. I hope this is available soon 👍

**Comment by Braintelligence** [comment]
@svengo Do you have any plans on further working on this?

File: docs/firststeps-rp.md (hunk 1 above)
Comment (MAGICCC) [reply docs/firststeps-rp.md]: I would remove the part with the org docker-compose.yml, so users need to use the override one

**Comment by svengo** [comment]
@Braintelligence I'm sorry, but I don't have any spare time at the moment to work on mailcow. :-(

**Comment by Braintelligence** [comment]
@svengo No problem! Just wanted to know if I can try this out anytime soon. Wish you the very best for your current endeavour.

**Comment by alexanderadam** [comment]
Is this example configuration still up to date?
I'm just asking because Traefik 2 came out in September and I remember that some things regarding the configuration changed.

…otherwise this PR is from 2017 so I guess even if it would be updated it probably wouldn't be reviewed and merged before Traefik 3 anyway. :wink:

**Comment by xFirestorm** [comment]
We managed to get mailcow working with traefik 2.0 including LetsEncrypt: https://github.com/mailcow/mailcow-dockerized/issues/667#issuecomment-566626026

**Comment by itsb** [comment]
any update on this? the traefik 2.0 configs are not working for me

**Comment by milkmaker** [comment]
This pull request has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs.

**Pull request closed without merging**